x-pack/plugins/infra/public/common/visualizations/lens/dashboards/asset_details/host/host_metric_charts.ts

```diff
@@ -11,207 +11,383 @@ import { hostLensFormulas } from '../../../../constants';
 import type { XYChartLayerParams } from '../../../../types';
 import { REFERENCE_LINE, XY_OVERRIDES } from '../../constants';
 
-type DataViewOrigin = 'logs' | 'metrics';
+export const DataViewOrigin = {
+  Logs: 'logs',
+  Metrics: 'metrics',
+} as const;
+export type DataViewOrigin = typeof DataViewOrigin[keyof typeof DataViewOrigin];
 
-export const hostMetricCharts: Array<
-  Pick<TypedLensByValueInput, 'id' | 'title' | 'overrides'> & {
-    dataViewOrigin: DataViewOrigin;
-    layers: XYChartLayerParams[];
-  }
-> = [
-  {
-    id: 'cpuUsage',
-    title: i18n.translate('xpack.infra.assetDetails.metricsCharts.cpuUsage', {
-      defaultMessage: 'CPU Usage',
-    }),
-
-    layers: [
-      {
-        data: [hostLensFormulas.cpuUsage],
-        type: 'visualization',
-      },
-    ],
-    dataViewOrigin: 'metrics',
-    overrides: {
-      axisLeft: XY_OVERRIDES.axisLeft,
+const LayerParamsType = {
+  Visualization: 'visualization',
+  ReferenceLines: 'referenceLines',
+} as const;
+
+const seriesType = {
+  Area: 'area',
+  AreaPercentageStacked: 'area_percentage_stacked',
+  AreaStacked: 'area_stacked',
+} as const;
+
+const TOP_VALUES_TYPE = 'top_values' as const;
```

Comment:
Would it be better to rely on types instead of defining these constants?

Say that we define a type that is:

```
type XYConfig =  Pick<TypedLensByValueInput, 'id' | 'title' | 'overrides'> & {
    dataViewOrigin: DataViewOrigin;
    layers: XYChartLayerParams[];
  }
```

we could do something like this:

```ts
const diskIOReadWrite: XYConfig = {
  id: 'diskIOReadWrite',
  title: i18n.translate('xpack.infra.assetDetails.metricsCharts.diskThroughput', {
    defaultMessage: 'Disk Throughput',
  }),
  layers: [
    {
      data: [
        {
          ...hostLensFormulas.diskReadThroughput,
          label: i18n.translate('xpack.infra.assetDetails.metricsCharts.metric.label.read', {
            defaultMessage: 'Read',
          }),
        },
        {
          ...hostLensFormulas.diskWriteThroughput,
          label: i18n.translate('xpack.infra.assetDetails.metricsCharts.metric.label.write', {
            defaultMessage: 'Write',
          }),
        },
      ],
      options: {
        seriesType: 'area',
      },
      type: 'visualization',
    },
  ],
  overrides: {
    settings: XY_OVERRIDES.settings,
  },
  dataViewOrigin: DataViewOrigin.Metrics,
};

This way we can ensure if the type change we'll see it reflected here.

Reply:
Nice idea, I like it! It makes more sense to do that. Thanks for the comment, changed.